### PR TITLE
Handle network from argument passed to start

### PR DESCRIPTION
**Pull request by pnowosie**
This PR resolves following JIRA ticket

 
- [APPS-150](https://golemproject.atlassian.net/browse/APPS-150)

### Why
- We need a way to configure the node (make it sync with the desired network)

### What
 - start method in ya-runtime-erigon should accept “config”/”params” JSON dictionary so far the only key being “network” with possible values: “mainnet”, “goerli”, “rinkeby”, “ropsten”, “kovan”
- Runtime should pass the network parameter to erigon.
- When config is not present, default values should be used (“goerli” in case of “network”).

### Assumptions 

[-] Latest [debugger](https://github.com/golemfactory/ya-runtime-dbg/pull/7) 

### How to test

```bash
ya-runtime-dbg \
  --runtime ./target/debug/ya-runtime-erigon \
  --workdir ./target/debug \
  --start-arg '{"network": "kovan"}'
```

### More
This PR also addresses renaming `ya-erigon-runtime` to `ya-runtime-erigon` (as in https://golemproject.atlassian.net/browse/APPS-152)

### CODE REVIEW

File: ya-runtime-erigon/src/main.rs [hunk 1]
```diff
@@ -129,8 +158,10 @@ impl Runtime for ErigonRuntime {
 
         self.erigon_pid = Some(erigon_pid);
         self.rpcdaemon_pid = Some(rpcd_pid);
+        self.erigon_chain_id = Some(chain_id.clone());
 
-        async move { Ok("start".into()) }.boxed_local()
+        let chain_id_copied = chain_id.clone();
+        async move { Ok(serialize::json::json!({ "network": chain_id_copied })) }.boxed_local()
```
Comment (Wiezzel) [inline ya-runtime-erigon/src/main.rs:164]: :question: 
```suggestion
        async move { Ok(serialize::json::json!({ "network": chain_id })) }.boxed_local()
```

File: ya-runtime-erigon/src/main.rs [hunk 2]
```diff
@@ -84,9 +103,13 @@ impl Runtime for ErigonRuntime {
     }
 
     fn start<'a>(&mut self, ctx: &mut Context<Self>) -> OutputResponse<'a> {
-        //TODO: Receive chain_id from start parameters when Yapapi ready
-        // see: https://github.com/golemfactory/yapapi/issues/390
-        let chain_id = String::from("goerli");
+        let default_chain_id = "goerli".to_string();
+        let chain_id = if let ya_runtime_sdk::cli::Command::Start { args } = &ctx.cli.command {
+            parse_json(args.into_iter().next(), &default_chain_id)
```
Comment (Wiezzel) [inline ya-runtime-erigon/src/main.rs:108]: Is really `args.into_iter().next()` the proper way to get command line params in `ya_runtime_sdk`? Looks hacky. Summon @mfranciszkiewicz

File: ya-runtime-erigon/src/main.rs [hunk 3]
```diff
@@ -246,3 +279,23 @@ fn generate_password_file(
         .stderr(Stdio::null())
         .spawn()
 }
+
+fn parse_json(config: Option<&String>, default_chain_id: &String) -> String {
```
Comment (Wiezzel) [inline ya-runtime-erigon/src/main.rs:283]: `parse_json` is not very informative. How about `get_chain_from_config`? Also, having `default_chain_id` as param is odd. Better use a module-level constant.

File: ya-runtime-erigon/src/main.rs [hunk 4]
```diff
@@ -246,3 +279,23 @@ fn generate_password_file(
...
+            let value: Value =
+                serde_json::from_str(json).expect("Cannot parse config, assumes json string");
+            let network = value["network"].as_str().unwrap();
+            let _ = chain_id_valid(network).unwrap();
+            network.to_owned()
+        }
+    }
+}
+
+fn chain_id_valid(chain_id: &str) -> Result<(), String> {
```
Comment (Wiezzel) [inline ya-runtime-erigon/src/main.rs:296]: If we use an enum type for chains/networks we can replace this with `EnumFromStr` macro from [enum_derive](https://docs.rs/enum_derive/0.1.7/enum_derive/) crate.

File: ya-runtime-erigon/src/main.rs [hunk 5]
```diff
@@ -56,12 +60,19 @@ pub struct ErigonRuntime {
...
+    erigon_chain_id: Option<String>,
 }
 
 impl Runtime for ErigonRuntime {
     const MODE: RuntimeMode = RuntimeMode::Server;
 
     fn deploy<'a>(&mut self, ctx: &mut Context<Self>) -> OutputResponse<'a> {
+        let data_dir = ctx.conf.data_dir.clone();
+        let chain_subdirs: Vec<PathBuf> = SUPPORTED_CHAINS
+            .into_iter()
```
Comment (Wiezzel) [inline ya-runtime-erigon/src/main.rs:72]: If we use an enum type for chains/networks we use here `IterVariantNames` macro from [enum_derive](https://docs.rs/enum_derive/0.1.7/enum_derive/) crate.

File: ya-runtime-erigon/src/main.rs [hunk 6]
```diff
@@ -14,14 +16,12 @@ const AUTH_ERIGON_USER: &str = "erigolem";
 const ERIGON_BIN: &str = "tg";
 const RPCDAEMON_BIN: &str = "rpcdaemon";
 
+const SUPPORTED_CHAINS: &[&str; 4] = &["goerli", "rinkeby", "ropsten", "kovan"];
```
Comment (Wiezzel) [inline ya-runtime-erigon/src/main.rs:19]: This looks like a good use case for an enum. :wink:

**Review by Wiezzel — CHANGES_REQUESTED** [review]
**Do not use** `_id` suffix for chains/networks. We're operating with *names* not *IDs* here.

File: ya-runtime-erigon/src/main.rs (hunk 2 above)
Comment (mfranciszkiewicz) [reply ya-runtime-erigon/src/main.rs:108]: @Wiezzel `args: Vec<String>`

Comment (pnowosie) [reply ya-runtime-erigon/src/main.rs:108]: It will change in [APPS-152](https://golemproject.atlassian.net/browse/APPS-152)

**Review by Wiezzel — CHANGES_REQUESTED** [review]
https://github.com/golemfactory/yagna-service-erigon/pull/20 :point_left: My comments as a PR.

**Review by Wiezzel — APPROVED** [review]

**Pull request merged**